utils/fixed_decimal/src/decimal.rs

```diff
@@ -2078,6 +2078,185 @@ impl FixedDecimal {
     }
 }
 
+/// A struct containing a [`FixedDecimal`] significand together with an exponent, representing a
+/// number written in scientific notation, such as 1.729×10³.
+/// This structure represents any 0s shown in the significand and exponent,
+/// and an optional sign for both the significand and the exponent.
+#[derive(Debug, Clone, PartialEq)]
+pub struct ScientificDecimal {
+    significand: FixedDecimal,
+    // Invariants:
+    // - exponent.lower_magnitude == 0 (exponent has no fractional part).
+    exponent: FixedDecimal,
+}
+
+impl ScientificDecimal {
+    pub fn from_parts(significand: FixedDecimal, exponent: FixedDecimal) -> Result<Self, Error> {
+        if exponent.lower_magnitude != 0 {
+            Err(Error::IntegerExponent)
+        } else {
+            Ok(ScientificDecimal {
+                significand,
+                exponent,
+            })
+        }
+    }
+}
+
+/// Render the [`ScientificDecimal`] as a string of ASCII digits with a possible decimal point,
+/// followed by the letter 'e', and the exponent.
+///
+/// # Examples
+///
+/// ```
+/// # use fixed_decimal::FixedDecimal;
+/// # use fixed_decimal::ScientificDecimal;
+/// # use std::str::FromStr;
+/// # use writeable::assert_writeable_eq;
+/// #
+/// assert_writeable_eq!(
+///     ScientificDecimal::from_parts(FixedDecimal::from_str("+1.729").unwrap(),
+///                                   FixedDecimal::from_str("+03").unwrap()).unwrap(),
+///     "+1.729e+03");
+/// ```
+impl writeable::Writeable for ScientificDecimal {
+    fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
+        self.significand.write_to(sink)?;
+        sink.write_char('e')?;
+        self.exponent.write_to(sink)
+    }
+
+    fn writeable_length_hint(&self) -> writeable::LengthHint {
+        self.significand.writeable_length_hint() + 1 + self.exponent.writeable_length_hint()
+    }
+}
+
+writeable::impl_display_with_writeable!(ScientificDecimal);
+
+impl FromStr for ScientificDecimal {
+    type Err = Error;
+    fn from_str(input_str: &str) -> Result<Self, Self::Err> {
+        Self::try_from(input_str.as_bytes())
+    }
+}
+
+impl TryFrom<&[u8]> for ScientificDecimal {
+    type Error = Error;
+    fn try_from(input_str: &[u8]) -> Result<Self, Self::Error> {
+        if let Some((significand_str, mut exponent_str)) = input_str
+            .iter()
+            .position(|&c| c == b'e')
```

Comment:
On the one hand this isn’t really sampleValue syntax (because that doesn’t support scientific notation, and thus doesn’t support a sign or leading 0s in the exponent), on the other hand I would like it to remain close to that, in case CLDR decides to support pluralization on scientific notation.

sampleValue syntax is lowercase-only, so I think this puts us in a better place should CLDR move in that direction.